media/segment_rw.go

```diff
@@ -0,0 +1,130 @@
+package media
+
+import (
+	"errors"
+	"io"
+	"sync"
+)
+
+type SegmentReaderConfig struct{}
+
+// NewSegmentWriter makes a ring of n MediaWriters.
+func NewSegmentWriter(n int) *SegmentWriter {
+	size := n + 1 // make space for 'next'
+	rw := &SegmentWriter{
+		writers: make([]*MediaWriter, size),
+		size:    size,
+		seq:     -1, // -1 makes logic simpler in first Next()
+	}
+	rw.writers[0] = NewMediaWriter() // precreate first segment
+	return rw
+}
+
+type SegmentWriter struct {
+	mu      sync.Mutex
+	writers []*MediaWriter
+	size    int
+	seq     int
+	closed  bool
+}
+
+type writerWrapper struct{ mw *MediaWriter }
```

Comment:
Lgtm, but just curious do we need this struct? It doesn't have any extra functions on top of `io.WriteCloser` I think?